### DISCUSSION TITLE
Editing a file without overwriting the original file

### DISCUSSION THREAD

**Discussion by eduressia** [ImageMagick/ImageMagick#6386] [Help]
### ImageMagick version

7.1.1-11 Q16 x64

### Operating system, version and so on

WIN 10 Pro 22H2

### Description

I've been trying for hours to figure it out how to write a simple batch file that resizes a bunch of images and keeps the original untouched
Here is what I need:
magick mogrify -resize 480  *.png  -outputFilename: *.tip.png
If I try 
magick mogrify -resize 480  *.png  *.tip.png
it throws an error saying that it can't find *.tip or something like that
I know I roughly can use:
[Code: batchMagick.bat]
mkdir size480
C:\\"Program Files"\\"ImageMagick-7.1.1-Q16"\\magick mogrifY -resize 480 -path size480/ *.png
cd size480
ren *.png *.tip.png
move *.* .. 
cd..
rmdir size480
exit
[/code]
But has to be an easier way isn't?
Please help, I read pretty much all parameters in the main website and half of the discussions here, some times is easier to just ask again if that's the case

**Comment by fmw42** [comment]
create a new directory for the outputs and use the -path argument in mogrify.  See https://imagemagick.org/Usage/basics/#mogrify

**Reply by eduressia** [reply]
That's exactly what I'm currently doing, I need the resized image in the same folder in order to other program can read it while I hover the mouse over the original image and shows me the amplified image to see it in detail. Thanks! When I can, I'll see to check the source code to add a new modifier -outpuFilename to rename the output without overwriting the original file, as the warning in the link you provided says "Mogrify is dangerous", lol

**Comment by snibgo** [comment]
`mogrify` doesn't use an output filename specified in the command. It loops through the input files, and creates one output file for each input. The output filename will be the same as the input filename, but with a different extension if `-format` is used, eg `-format PNG` will use a ".png" extension.

`mogrify` can use a `-path`, which will be the directory for the output files. Then you can use shell commands to move or copy the output files to the current directory.

An alternative is to create a shell `for` loop to process each input file individually. That gives you flexibility to name the outputs differently to the inputs, but will be slightly slower as you will call `magick` multiple times.

Another alternative is to have a single `magick` command that reads all your inputs, then writes them all out. This also gives you flexibility for naming outputs. However, if the total input files have more pixels than will fit into your memory, this will be slow, or may fail.

**Reply by eduressia** [reply]
Thanks, I was checking the source code, I noticed all you told me, I don't have much time nor the tools to compile it (yet), but seems doable nevertheless, I'll give it a try, "Ït's just an array,  just pull and push until it fits", lol

**Comment by msypien-kg** [comment]
@eduressia, with `morify`, you can use the `-write` flag. For example, I like to combine it with `find`, `parallel` to speed up the transformation process and save it to another directory. Later, you can rename files if you want.

```
mkdir myNewDir
find . -name "*.png" -maxdepth 1 -type f | parallel magick mogrify -resize 480x480 -write "myNewDir/{}" {}
```

**Comment by ShallowSpace** [comment]
Use this code
**Magick  mogrify -path "Folder 02"   +repage     -resize 1x1      +repage    "Folder 01"/*jpg**
*It takes the files from folder "Folder 01" and places the modified files into folder "Folder 02". Obviously make sure that "Folder 02" exists. 
*Note that I put "" marks around the folder name because the folder name has a space mark. Without the "" the system will look for "01" not "Folder 01"
*I put "+repage" to prevent errors. Its may not be not needed in your case. 

*Once you have all of the files in Folder 02 _**try**_ to place them back in Folder 01. The system will ask if you want to replace or "Compare info for both files". Click on "Compare..." then check the tick box for both folders. This will rename the incoming files. So your folder will have "image.jpg" and it will also have the newly resized "image (2).jpg"

Again: the steps are 
1) Make sure "Folder 02" Exists
2) Run command
3) Dump files from Folder 02 into Folder 01. A window will pop up. 
4) On the new window click "Compare...". Then tick both boxes. Then click continue. 
Done
 
Note that this works for many other commands like cropping, color changing, rotating, etc.
